src/pages/workspaces/workspace/Data.js

```diff
@@ -563,17 +566,35 @@ const WorkspaceData = _.flow(
     }
   }
 
-  const loadWdsSchema = async () => {
-    if (isFeaturePreviewEnabled('workspace-data-service') && !getConfig().isProd) {
-      try {
-        setWdsSchema([])
-        setWdsSchemaError(undefined)
-        const wdsSchema = await Ajax(signal).WorkspaceData.getSchema(workspaceId)
-        setWdsSchema(wdsSchema)
-      } catch (error) {
-        setWdsSchemaError(error)
+  const loadWdsSchemaAndUrl = async () => {
+    await Ajax(signal).Apps.getV2AppInfo(workspaceId).then(async apps => {
+      // TODO: Update to pull specifically from /api/apps/v2/{workspaceId}/{appName}
+      if (isFeaturePreviewEnabled('workspace-data-service') && !getConfig().isProd) {
+        try {
+          setWdsSchema([])
+          setWdsSchemaError(undefined)
+          const url = getWdsUrl(apps)
+          setWdsProxyUrl(url)
+          const wdsSchema = await Ajax(signal).WorkspaceData.getSchema(url, workspaceId)
+          setWdsSchema(wdsSchema)
+        } catch (error) {
+          setWdsSchemaError(error)
+        }
       }
+    })
+  }
+
+  // TODO: Do we have the appName here yet? If so, we could just call /api/apps/v2/{workspaceId}/{appName}?
+  const getWdsUrl = apps => {
+    // TODO: determine appropriate logic
+    const candidates = apps.filter(app => app.appType === 'CROMWELL' && app.status === 'RUNNING')
+    if (candidates.length === 0) {
+      //TODO: panic
+    }
+    if (candidates.length > 1) {
+      candidates.sort((a, b) => a.auditInfo.createdDate - b.auditInfo.createdDate)
     }
+    return candidates[0].proxyUrls.wds
   }
```

Comment:
I think we should keep calling the `listAppsV2`, instead of calling `getAppV2` and specifying an expected app name; but this function can encode the expected appName right into itself:

```suggestion
  const getWdsUrl = apps => {
    // look explicitly for an app named 'cbas-wds-default'. If found, use it, even if it isn't running
    // this handles the case where the user has explicitly shut down the app
    const namedApp = apps.filter(app => app.appType === 'CROMWELL' && app.appName === 'cbas-wds-default')
    if (namedApp.length === 1) {
      return namedApp[0].proxyUrls.wds
    }
    // if we didn't find the expected app 'cbas-wds-default', go hunting:
    const candidates = apps.filter(app => app.appType === 'CROMWELL' && app.status === 'RUNNING')
    if (candidates.length === 0) {
      // no app deployed yet
      return ''
    }
    if (candidates.length > 1) {
      // multiple apps found; use the earliest-created one
      candidates.sort((a, b) => a.auditInfo.createdDate - b.auditInfo.createdDate)
    }
    return candidates[0].proxyUrls.wds
  }
```

Reply:
Just curious, why `cbas-wds-default` specifically? I'm trying to find if there is code where that might be the default value for an app without supplying one? Unless this is a value we would like to use?

Reply:
I'm just picking a name arbitrarily, and when we do the follow-on work to handle deployments, hopefully we can use this name we picked. Feel free to choose a different name! In the worst case, when we start deploying, we can revise this code to reflect the actual name.